### PR TITLE
Fixed #33419 -- Restored marking forms.Field.help_text as HTML safe.

### PR DESCRIPTION
**Pull request by smithdc1**
[Ticket 33419](https://code.djangoproject.com/ticket/33419#comment:2)

### CODE REVIEW

File: tests/forms_tests/tests/test_forms.py [hunk 1]
```diff
@@ -2326,6 +2326,34 @@ class UserRegistration(Form):
 <input type="hidden" name="next" value="/"></li>"""
         )
 
+    def test_help_text_is_escaped(self):
```
Comment (felixxm) [inline tests/forms_tests/tests/test_forms.py]:
```suggestion
    def test_help_text_html_safe(self):
```

File: docs/releases/4.0.2.txt [hunk 2]
```diff
@@ -11,3 +11,6 @@ Bugfixes
 
 * Fixed a bug in Django 4.0 where ``TestCase.captureOnCommitCallbacks()`` could
   execute callbacks multiple times (:ticket:`33410`).
+
+* Fixed a bug in Django 4.0 where ``help_text`` was not escaped in form
```
Comment (felixxm) [inline docs/releases/4.0.2.txt]: `was not escaped` -> `was escaped`

**Review by felixxm — COMMENTED** [review]
@smithdc1Thanks :+1: I pushed small edits to release notes.

File: docs/releases/4.0.2.txt (hunk 2 above)
Comment (smithdc1) [reply docs/releases/4.0.2.txt]: Your wording here is much better. Thank you.

File: tests/forms_tests/tests/test_forms.py (hunk 1 above)
Comment (smithdc1) [reply tests/forms_tests/tests/test_forms.py]: Yes, I got my this phrasing here the wrong way round. Reading the test name again, I'm not sure it's that obvious what is going on here. Maybe we should add a doc string? I could also be overthinking it at this point. 
```python
"""
help_text is provided by developers and therefore should not be escaped. See ticket #6041.
"""
```

Comment (felixxm) [reply tests/forms_tests/tests/test_forms.py]: I added a short docstring but without `is provided by developers ...`, this is not a strong argument, a lot of things are provided by developers but we don't mark them as safe. I'd say that's how it is, the decision was made a lot time ago :shrug:

**Pull request merged**